src/common/utility.ts

```diff
@@ -475,17 +475,18 @@ export class Utility {
         return deployment;
     }
 
-    public static serializeCreateOptions(settings: any, createOptions: any): any {
+    public static serializeCreateOptions(moduleKey: string, settings: any, createOptions: any): any {
         let optionStr: string;
         if (typeof createOptions === "string") {
-            optionStr = createOptions;
+            // remove line breaks and extra spaces
+            optionStr = JSON.stringify(JSON.parse(createOptions), null, 0);
         } else {
             optionStr = JSON.stringify(createOptions);
         }
         const re = new RegExp(`.{1,${Constants.TwinValueMaxSize}}`, "g");
         const options = optionStr.match(re);
         if (options.length > Constants.TwinValueMaxChunks) {
-            throw new Error("Size of createOptions too big. The maxium size of createOptions is 4K");
+            throw new Error(`Size of createOptions too big in module ${moduleKey}. The maximum size of createOptions is 4K`);

```

Comment:
since we have settings in this method. We could get image name from settings. So no need to pass module.